gitlab.go

```diff
@@ -187,9 +187,6 @@ func (c *Client) SetBaseURL(urlStr string) error {
 		return err
 	}
 
-	// Set the encoded opaque data
-	c.baseURL.Opaque = c.baseURL.Path
-
```

Comment:
URL was changed in Go 1.5 and this workaround with Opaque is not needed. (See: https://github.com/golang/go/commit/874a605af0764a8f340c3de65406963f514e21bc).
I did small fix and added additional test for finding project by `NAMESPACE/PROJECT`. It's working in Go 1.5.
Now I realized that travis job is configured to use Go 1.4. This is the reason why test is failing. Do we want to stay at 1.4 or can we upgrade to 1.5?